### PR TITLE
feature: Add development guide

### PR DESCRIPTION
**Pull request by tore-espressif**
Add development guide

### CODE REVIEW

File: BSP development guide.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Board Support Package development guide
+
+This guide describes necessary steps add a new BSP to the esp-bsp project.
```
Comment (VojtechBartoska) [inline BSP development guide.md:3]: **to** add

File: BSP development guide.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+* We want to provide LVGL users with easy, out-of-the-box working solution for their GUI applications. Each board that has a display must be added to LVGL's [SquareLineStudio](https://squareline.io/) pre-defined boards. For more information refer to [this readme](https://github.com/espressif/esp-bsp/blob/master/SquareLine/README.md).
+    * List of currently implemented LCD drivers is maintained in a [separate table](https://github.com/espressif/esp-bsp/blob/master/LCD.md)
+* We have an experimental idf.py extension, [bsp_ext.py](https://github.com/espressif/esp-bsp/blob/master/examples/bsp_ext.py) that allows changing of BSP during project configuration. Please add new BSPs to the list in this file. 
+
+### API conventions
+
+BSP is a standard IDF component. You can start development by `idf.py create-component your_bsp_name`.
+
+The APIs that a BSP can offer depend on the boards capabilities. For example, a board that contains audio codec will expose audio API and a board with display and touch controller will expose display API.
```
Comment (VojtechBartoska) [inline BSP development guide.md:18]: Missing mention of touch API at the end of this paragraph: "will expose display and **touch** API."

File: BSP development guide.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+#define BSP_CAPS_AUDIO_MIC      1
+#define BSP_CAPS_SDCARD         0
+#define BSP_CAPS_IMU            1
+```
+* Each BSP must provide a wrapper header file that can be included by `#include "bsp/esp-bsp.h"`
+* BSPs that expose display and touch APIs must provide a header file `bsp/display.h` and `bsp/touch.h` that contain functions for display/touch initialization without graphical library (LVGL)
+* Complete board's pinout must be in the beginning of the BSP's header file
+
+#### I2C
+* I2C can used by many devices on the bus. The 'init' call must support multiple invocations
```
Comment (VojtechBartoska) [inline BSP development guide.md:41]: I2S can **be**

**Review by VojtechBartoska — COMMENTED** [review]
LGTM, just few typos

**Comment by tore-espressif** [comment]
@espzav PTAL! Only nitpick changes since your last review!

**Review by espzav — APPROVED** [review]
@tore-espressif Thank you! LGTM

**Pull request merged**